api/management/api.go

```diff
@@ -584,30 +584,39 @@ func (rac RealmAdminConfiguration) Validate() error {
 	return validation.NewParameterValidator().
 		ValidateParameterIn("mode", rac.Mode, allowedAdminConfMode, true).
 		ValidateParameterFunc(rac.validateAvailableChecks).
-		ValidateParameterFunc(rac.validateAccreditations).
 		Status()
 }
 
 func (rac RealmAdminConfiguration) validateAvailableChecks() error {
+	var accredConditions, err = rac.validateAccreditations()
+	if err != nil {
+		return err
+	}
+
 	if len(rac.AvailableChecks) > 0 {
-		for k := range rac.AvailableChecks {
+		for k, v := range rac.AvailableChecks {
 			if !validation.IsStringInSlice(configuration.AvailableCheckKeys, k) {
 				return errorhandler.CreateBadRequestError(constants.MsgErrInvalidParam + ".available-checks")
 			}
+			if _, ok := accredConditions[k]; v && !ok {
+				return errorhandler.CreateBadRequestError(constants.MsgErrMissingParam + ".accreditations." + k)
+			}
 		}
 	}
 	return nil
 }
 
-func (rac RealmAdminConfiguration) validateAccreditations() error {
+func (rac RealmAdminConfiguration) validateAccreditations() (map[string]bool, error) {
+	var accredConditions = make(map[string]bool)
 	if len(rac.Accreditations) > 0 {
```

Comment:
Est ce que le check sur length est vraiment nécessaire. Si len == 0 , le for ne vas pas tourner non ?